csv-enrichment/vector.toml

```
@@ -15,15 +32,40 @@ lines = [
 ]
 interval = 2
 
-[transforms.remap]
+# Generate coded error messages
+[sources.coded_error_messages]
+type = "generator"
+format = "shuffle"
+lines = [
+  '{"code":1,"device_id":"e5ad503d","timestamp":"2021-10-18T15:35:09.158139Z"}',
+  '{"code":2,"device_id":"a5b2401e","timestamp":"2021-10-18T15:35:28.517210Z"}',
+  '{"code":3,"device_id":"b48f41aa","timestamp":"2021-10-18T15:35:37.846783Z"}'
+]
+interval = 2
+
+[transforms.remap_user_info]
 type = "remap"
-inputs = ["random"]
+inputs = ["user_info_messages"]
 source = """
 . = parse_json(.message) ?? {}
 . |= get_enrichment_table_record("users", { "last_name": .last_name, "first_name": .first_name }) ?? {}
 """
 
+[transforms.remap_coded_errors]
+type = "remap"
+inputs = ["coded_error_messages"]
+source = """
+. = parse_json(.message) ?? {}
```

```suggestion
. = parse_json!(.message)
```

I don't see any reason to carry on processing if `parse_json` fails.